pkg/virt-controller/watch/vm.go

```diff
@@ -217,6 +220,30 @@ func (c *VMController) execute(key string) error {
 		}
 	}
 
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.VolumeSource.PersistentVolumeClaim != nil {
+			key := fmt.Sprintf("%s/%s", vm.Namespace, volume.VolumeSource.PersistentVolumeClaim.ClaimName)
+			obj, exists, err := c.pvcInformer.GetStore().GetByKey(key)
+			if exists && err == nil {
+				pvc := obj.(*k8score.PersistentVolumeClaim)
+				for _, or := range pvc.ObjectMeta.OwnerReferences {
+					if or.Kind == "DataVolume" {
```

Comment:
Is there something like an annotation on a PVC which indicates that the import is done? From my perspective the user-friendly solution would be if people do not have to care about the origin of a PVC and can just attach one, if they see it.

Reply:
We can then basically just wait until the import is done and then continue with the pod creation.

Reply:
Also looking up the datavolume based on the owner reference would work perfectly fine.

Reply:
I would shy away from having KubeVirt know about internals of CDI. We can (and have) modified the annotation we use to indicate certain things, including success. The problem this is solving is that someone used a DataVolume to populate the disk, but then in the VM definition mistakenly used the underlying PVC bypassing the check to see if the DataVolume is completed. Since it is perfectly valid if someone created their own PVC with a disk image on it somehow, I would not make the check generic for PVCs. Only for PVCs that have a ownerRef of a DataVolume (which is what this code does)

Reply:
After the upload is done, rlthe pvc is like any other pvc. Why can't we check the owning Datavolume and continue when it is done?

Reply:
We were confused on what to do. We had 2 options:
1. Warn the user like this code is doing and continue on as before
2. Block the VM from starting as if we are using a DV.
After looking it over some, I would agree with you, we should look up the DV using the ownerReference, and then block the VM as if we are using a DV instead of PVC.

Reply:
Great, I think option 2is by far the most user friendly solution. 👍

Reply:
Cool. I'm on it.

Reply:
I am going to disagree with the premise that we can automatically treat a kubevirt persistentVolumeClaim volume type like a dataVolume volumeType when we find an ownerReference.

Sure it works now but we want to introduce special WaitForFirstConsumer handling for DataVolumes in kubevirt in the future which will require additional interaction with the DataVolume on kubevirt's side.  It's not clear to me how we could (or if we would want to) implicitly handle that too.

This only works accidentally because the pvc and dv have the same name.

In my view, using a DataVolume as a persistentVolumeClaim type is an error in the spec and should be handled as such.  Are there any other scenarios where you "correct" the user and change the semantics of a declared state because kubevirt knows better than the user?

Reply:
This all stems from the fact that we use the same name for the dv and pvc.  When the user uses the wrong volume type with a DV it works by dumb luck.  If the pvc and dv names were different the expected (and correct) behavior would be the virt-launcher pod failing to start because the requested PVC does not exist.  

If we want to have correct behavior, as I see it, we can do two things:
1. Fix CDI to change the name of the pvc so it is different from the dv name
2. Add the ownerReference check in kubevirt and refuse to start the VM

We want to do 1 but it could be invasive.  It would be nice to have a fix sooner.  I would rather keep the current broken behavior than paper over the problem with an implicit behavior as has been suggested by @rmohr .

Reply:
I don't understand what you mean. There is an owner reference. If the owner says it is done, then it is done. No matter if Pvc names match with anything or not. 

I would not support anything like wait WaitFirFistConsumer with Datavolumes. Datavolumes are messy enough already. I would only support transparent imports on a PV basis with such featured. With the current model something like WaitForFistConsumer dies also not work with Datavolumes directly referenced

Reply:
And even if we add features like this, at the end one always has to look on the Datavolume. Done via owner reference or directly. Where is the difference?

Reply:
At the end the goal always was to bring importing/cloning to PVs and Datavolumes are just a hack to get that. Right?

Reply:
Please go ahead with option 1 so that we can at least inform people that it will not work, before we have no movement. But this is really a pain for users.

Reply:
They still can use PVCs directly if they want to self-manage the import process.  If they want managed imports then they currently must use DataVolumes.  Let's keep the two cases distinct and not try to infer what the user wants.

Reply:
Took me some time, but now I see and agree. Sorry for the noise.